### ISSUE TITLE
Training config and the training time  for OCR result

### ISSUE THREAD

**Issue by Yovecent** [yifan123/flow_grpo#96]
Dear Author, I'm trying to train the sd3 model to reproduce the OCR result in your paper.
I just want to know how many time will cost to get OCR result.
Like, If I have 4 nodes, each node with 8 gpu to run the below code.

```
# Master node
bash scripts/multi_node/sd3/main.sh
# Other nodes
bash scripts/multi_node/sd3/main1.sh
bash scripts/multi_node/sd3/main2.sh
bash scripts/multi_node/sd3/main3.sh    # change the config to  general_ocr_sd3.py
```

how many time will it cost to get the correct result?

And I also want to know what's your training config to get the OCR result.

**Comment by Yovecent** [comment]
> The training config is in https://github.com/yifan123/flow_grpo/blob/main/config/grpo.py#L83
> 
> With the above config, it reaches 90% accuracy at 1k steps, requiring 16 hours on 24 A-series GPUs or 8 hours on 24 H-series GPUs.

I was trying to train 4 node (each 8 gpu  40G  A100) on OCR result in below config.
But I meet the OOM problem.



How can I change the `train_batch_size`  and `num_image_per_prompt ` to obmit OOM problem

```
def general_ocr_sd3():
    gpu_number = 32
    config = compressibility()
    config.dataset = os.path.join(os.getcwd(), "dataset/ocr")

    # sd3.5 medium
    config.pretrained.model = "/share/project/denghaoge/wangjiaqi/data/model_ckpt/stable-diffusion-3.5-medium"
    config.sample.num_steps = 10
    config.sample.eval_num_steps = 40
    config.sample.guidance_scale = 4.5

    config.resolution = 512
    config.sample.train_batch_size = 3
    config.sample.num_image_per_prompt = 4
    config.sample.num_batches_per_epoch = int(48/(gpu_number*config.sample.train_batch_size/config.sample.num_image_per_prompt))
    assert config.sample.num_batches_per_epoch % 2 == 0, "Please set config.sample.num_batches_per_epoch to an even number! This ensures that config.train.gradient_accumulation_steps = config.sample.num_batches_per_epoch / 2, so that gradients are updated twice per epoch."
    config.sample.test_batch_size = 16 # 16 is a special design, the test set has a total of 1018, to make 8*16*n as close as possible to 1018, because when the number of samples cannot be divided evenly by the number of cards, multi-card will fill the last batch to ensure each card has the same number of samples, affecting gradient synchronization.

    config.train.batch_size = config.sample.train_batch_size
    config.train.gradient_accumulation_steps = config.sample.num_batches_per_epoch//2
    config.train.num_inner_epochs = 1
    config.train.timestep_fraction = 0.99
    # kl loss
    config.train.beta = 0.04
    # Whether to use the std of all samples or the current group's.
    config.sample.global_std = True
    # Whether to use the same noise for the same prompt
    config.sample.same_latent = False
    config.train.ema = True
    # A large num_epochs is intentionally set here. Training will be manually stopped once sufficient
    config.save_freq = 60 # epoch
    config.eval_freq = 60
    config.save_dir = 'logs/ocr/sd3.5-M'
    config.reward_fn = {
        "ocr": 1.0,
    }
    
    config.prompt_fn = "general_ocr"

    config.per_prompt_stat_tracking = True
    return config
```

**Comment by Yovecent** [comment]
> > The training config is in https://github.com/yifan123/flow_grpo/blob/main/config/grpo.py#L83
> > With the above config, it reaches 90% accuracy at 1k steps, requiring 16 hours on 24 A-series GPUs or 8 hours on 24 H-series GPUs.
> 
> I was trying to train 4 node (each 8 gpu 40G A100) on OCR result in below config. But I meet the OOM problem.
> 
> How can I change the `train_batch_size` and `num_image_per_prompt ` to obmit OOM problem
> 
> ```
> def general_ocr_sd3():
>     gpu_number = 32
>     config = compressibility()
>     config.dataset = os.path.join(os.getcwd(), "dataset/ocr")
> 
>     # sd3.5 medium
>     config.pretrained.model = "/share/project/denghaoge/wangjiaqi/data/model_ckpt/stable-diffusion-3.5-medium"
>     config.sample.num_steps = 10
>     config.sample.eval_num_steps = 40
>     config.sample.guidance_scale = 4.5
> 
>     config.resolution = 512
>     config.sample.train_batch_size = 3
>     config.sample.num_image_per_prompt = 4
>     config.sample.num_batches_per_epoch = int(48/(gpu_number*config.sample.train_batch_size/config.sample.num_image_per_prompt))
>     assert config.sample.num_batches_per_epoch % 2 == 0, "Please set config.sample.num_batches_per_epoch to an even number! This ensures that config.train.gradient_accumulation_steps = config.sample.num_batches_per_epoch / 2, so that gradients are updated twice per epoch."
>     config.sample.test_batch_size = 16 # 16 is a special design, the test set has a total of 1018, to make 8*16*n as close as possible to 1018, because when the number of samples cannot be divided evenly by the number of cards, multi-card will fill the last batch to ensure each card has the same number of samples, affecting gradient synchronization.
> 
>     config.train.batch_size = config.sample.train_batch_size
>     config.train.gradient_accumulation_steps = config.sample.num_batches_per_epoch//2
>     config.train.num_inner_epochs = 1
>     config.train.timestep_fraction = 0.99
>     # kl loss
>     config.train.beta = 0.04
>     # Whether to use the std of all samples or the current group's.
>     config.sample.global_std = True
>     # Whether to use the same noise for the same prompt
>     config.sample.same_latent = False
>     config.train.ema = True
>     # A large num_epochs is intentionally set here. Training will be manually stopped once sufficient
>     config.save_freq = 60 # epoch
>     config.eval_freq = 60
>     config.save_dir = 'logs/ocr/sd3.5-M'
>     config.reward_fn = {
>         "ocr": 1.0,
>     }
>     
>     config.prompt_fn = "general_ocr"
> 
>     config.per_prompt_stat_tracking = True
>     return config
> ```

I  change the code below



```
config.sample.train_batch_size = 1
config.sample.num_image_per_prompt = 1
config.sample.num_batches_per_epoch = 2
assert config.sample.num_batches_per_epoch % 2 == 0, "Please set config.sample.num_batches_per_epoch to an even number! This ensures that config.train.gradient_accumulation_steps = config.sample.num_batches_per_epoch / 2, so that gradients are updated twice per epoch."
config.sample.test_batch_size = 16 # 16 is a special design, the test set has a total of 1018, to make 8*16*n as close as possible to 1018, because when the number of samples cannot be divided evenly by the number of cards, multi-card will fill the last batch to ensure each card has the same number of samples, affecting gradient synchronization.

```


<img width="650" height="740" alt="Image" src="https://github.com/user-attachments/assets/039f209c-b96a-451b-bd2c-9983ca4f0599" />



What is the most perfect config for it?

**Comment by Yovecent** [comment]
> config.sample.num_image_per_prompt is the group size, please use config.sample.num_image_per_prompt = 8

do you mean this ?

```
config.sample.train_batch_size = 1
config.sample.num_image_per_prompt = 8
config.sample.num_batches_per_epoch = 2
assert config.sample.num_batches_per_epoch % 2 == 0, "Please set config.sample.num_batches_per_epoch to an even number! This ensures that config.train.gradient_accumulation_steps = config.sample.num_batches_per_epoch / 2, so that gradients are updated twice per epoch."
config.sample.test_batch_size = 16 # 16 is a special design, the test set has a total of 1018, to make 8*16*n as close as possible to 1018, because when the number of samples cannot be divided evenly by the number of cards, multi-card will fill the last batch to ensure each card has the same number of samples, affecting gradient synchronization.
```

**Comment by Yovecent** [comment]
> gpu_number = 32 config.sample.train_batch_size = 1 config.sample.num_image_per_prompt = 8 config.sample.num_batches_per_epoch =int(48/(gpu_number*config.sample.train_batch_size/config.sample.num_image_per_prompt))

Thanks a lot! It works!


So is `config.sample.train_batch_size` is the only factor to influence the Memory in one GPU  in Sampling Stage?


But for the training stage, what's the factor to influence the Memory?
